### PR TITLE
Build ES6 module

### PR DESCRIPTION
**Pull request by b1f6c1c4**
For those who prefer ES6 import/export (such as rollup and webpack user), it'd be better not to babel `transform-es2015-modules-commonjs`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,8 @@
     "url": "https://github.com/atlassian/react-beautiful-dnd/issues"
   },
   "main": "lib/index.js",
+  "module": "lib/es/index.js",
```
Comment (TrySound) [inline package.json]: Can you call it `esm`?

File: package.json [hunk 2]
```diff
@@ -26,8 +28,9 @@
     "validate": "yarn run lint && yarn run typecheck",
     "lint": "eslint .",
     "typecheck": "flow check",
-    "build": "yarn run build:clean && yarn run build:lib && yarn run build:flow && yarn run build:dist",
+    "build": "yarn run build:clean && yarn run build:lib && yarn run build:flow && yarn run build:dist && yarn run build:es",
     "build:clean": "rimraf lib && rimraf dist",
+    "build:es": "cross-env NODE_ENV=production BABEL_ENV=es babel ./src --out-dir ./lib/es",
```
Comment (TrySound) [inline package.json]: `esm` and babel_env is not required here

File: .babelrc [hunk 3]
```diff
@@ -12,6 +12,10 @@
   ],
   "comments": false,
   "env": {
+    "es": {
```
Comment (TrySound) [inline .babelrc]: this is not required

**Comment by b1f6c1c4** [comment]
@TrySound BTW, should we assume that those 'lib/es' users are able to accept ES6 so `transform-runtime`, `transform-class-properties`, `transform-object-rest-spread`, `transform-export-extensions` are unnecessary for them?

**Comment by TrySound** [comment]
@b1f6c1c4 Nope. That's why I ask you to call it `esm`.

File: package.json (hunk 2 above)
Comment (b1f6c1c4) [reply package.json]: To my knowledge babel uses `BABEL_ENV` to match the `env` section in `.babelrc`. It's possible to rename it here to `BABEL_ENV=esm` (and modify in `.babelrc`) but it won't work if we remove `BABEL_ENV`.

Comment (TrySound) [reply package.json]: I meant rename task to esm and remove BABEL_ENV which is not used.

Comment (TrySound) [reply package.json]: Btw production is not required here too

**Comment by b1f6c1c4** [comment]
@TrySound Just a kind reminder that in the 1027 packages used by my project, `es` is much more popular than `esm`. And (to my knowledge) they use ES5 syntax with ES6 import/exports.
```
history/es
i18next/dist/es
i18next-browser-languagedetector/dist/es
jquery-i18next/dist/es
material-ui/es
react-beautiful-dnd/lib/es
react-redux/es
react-router/es
react-router-dom/es
react-router-redux/es
recompose/es
redux/es
redux-form/es
redux-form-material-ui/es
redux-saga/es
redux-thunk/es
reselect/es
symbol-observable/es

acorn/dist/acorn_loose.es.js
acorn/dist/walk.es.js
acorn/dist/acorn.es.js
acorn/dist/acorn_loose.es.js
acorn/dist/walk.es.js
brcast/dist/brcast.es.js
deep-diff/index.es.js
immutable/dist/immutable.es.js
material-ui/index.es.js
material-ui-icons/index.es.js
node-fetch/lib/index.es.js
react-intl/lib/index.es.js
react-side-effect/lib/index.es.js

popper.js/dist/esm
theming/dist/esm
```

**Comment by TrySound** [comment]
It doesn't matter what "they" use. From users feedback I got that `es` often confuses. So it's much better to rely [@std/esm](https://github.com/standard-things/esm) name.

And you asked about removing `transform-runtime`, `transform-class-properties`, `transform-object-rest-spread`, `transform-export-extensions`, but they are converts to what any users able to consume.

File: package.json (hunk 1 above)
Comment (TrySound) [reply package.json]: also I'd place it in `esm/index.js`

File: package.json [hunk 4]
```diff
@@ -16,6 +16,8 @@
     "url": "https://github.com/atlassian/react-beautiful-dnd/issues"
   },
   "main": "lib/index.js",
+  "module": "esm/index.js",
+  "sideEffects": false,
   "files": [
     "/lib",
```
Comment (TrySound) [inline package.json]: also add esm here. It's npm filter

**Comment by TrySound** [comment]
ignore esm here https://github.com/atlassian/react-beautiful-dnd/blob/master/.eslintignore

**Comment by alexreardon** [comment]
This looks good to me! Thanks @b1f6c1c4 for raising this and seeing it through. Also thanks @TrySound for jumping on this one and providing a great review.

dance-3

**Comment by lukebatchelor** [comment]
Sweet, this looks awesome. This closes #235 (which has been sitting assigned to me for a while 😨).

Thanks for the input @TrySound, I had the same question about whether `es` should be "es5 with es6 imports" and whether you should be transpiling away classes, spread, etc. So thanks for clearing that up, TIL.

**Comment by alexreardon** [comment]
This will go out in 6.0 which will be shipping soon

**Pull request merged**